File: Python/Car.py
from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, plate_number, brand, year, color, num_seats, num_doors):
        # Konstruktor kelas Car, mewarisi atribut dari kelas Vehicle
        super().__init__(plate_number, brand, year, color)
        self.__num_seats = num_seats    # Inisialisasi jumlah kursi
        self.__num_doors = num_doors    # Inisialisasi jumlah pintu

    def get_num_seats(self):
        # Metode untuk mendapatkan jumlah kursi
        return self.__num_seats

    def set_num_seats(self, num_seats):
        # Metode untuk mengatur jumlah kursi
        self.__num_seats = num_seats

    def get_num_doors(self):
        # Metode untuk mendapatkan jumlah pintu
        return self.__num_doors

    def set_num_doors(self, num_doors):
        # Metode untuk mengatur jumlah pintu
        self.__num_doors = num_doors

File: cpp2/Car.cpp
#include <iostream>
#include <string>
#include <vector>
#include <bits/stdc++.h>
#include "Vehicle.cpp" // Include header file for base class

class Car : public Vehicle {
private:
    int num_seats; // Number of seats attribute
    int num_doors; // Number of doors attribute

public:
    // Constructor for Car class, inheriting attributes from Vehicle class
    Car(std::string plate_number, std::string brand, int year, std::string color, int num_seats, int num_doors) 
        : Vehicle(plate_number, brand, year, color), num_seats(num_seats), num_doors(num_doors) {}

    // Getter method for number of seats
    int get_num_seats() {
        return num_seats;
    }

    // Setter method for number of seats
    void set_num_seats(int num_seats) {
        this->num_seats = num_seats;
    }

    // Getter method for number of doors
    int get_num_doors() {
        return num_doors;
    }

    // Setter method for number of doors
    void set_num_doors(int num_doors) {
        this->num_doors = num_doors;
    }
};
